xds/internal/xdsclient/xds.go

```diff
@@ -561,6 +572,35 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 	return routesRet, nil
 }
 
+func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy) ([]*HashPolicy, error) {
+	var hashPoliciesRet []*HashPolicy
+	for _, p := range policies {
+		policy := HashPolicy{Terminal: p.Terminal}
+		switch p.GetPolicySpecifier().(type) {
+		case *v3routepb.RouteAction_HashPolicy_Header_:
+			policy.HashPolicyType = HashPolicyTypeHeader
+			policy.HeaderName = p.GetHeader().GetHeaderName()
+			regex := p.GetHeader().GetRegexRewrite().GetPattern().GetRegex()
+			re, err := regexp.Compile(regex)
+			if err != nil {
+				return nil, fmt.Errorf("hash policy %+v contains an invalid regex %q", p, regex)
+			}
+			policy.Regex = re
+			policy.RegexSubstitution = p.GetHeader().GetRegexRewrite().GetSubstitution()
+		case *v3routepb.RouteAction_HashPolicy_FilterState_:
+			if p.GetFilterState().GetKey() != "io.grpc.channel_id" {
+				return nil, fmt.Errorf("hash policy %+v contains an invalid key for filter state policy %q", p, p.GetFilterState().GetKey())
+			}
+			policy.HashPolicyType = HashPolicyTypeChannelID
+		default:
+			return nil, fmt.Errorf("hash policy %T is an unsupported hash policy", p.GetPolicySpecifier())
```

Comment:
As you mentioned in chat, this shouldn't NACK.
Should be OK to just log something and continue

Reply:
Changed to logger.